core/src/main/java/hudson/cli/SetBuildDescriptionCommand.java

```diff
@@ -12,6 +12,8 @@
 @Extension
 public class SetBuildDescriptionCommand extends CLICommand implements Serializable {
 
+    private static final long serialVersionUID = 2690066739899830515L;
```

Comment:
Not sure it is actually needed with the current synchronization methods

Reply:
As it implements Serializable I think it should have a number like that. If another serializatzion method is used, the class should not implement Serializable.

From the [documentation of Java 11](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html):
If a serializable class does not explicitly declare a serialVersionUID, then the serialization runtime will calculate a default serialVersionUID value for that class based on various aspects of the class, as described in the Java(TM) Object Serialization Specification. However, it is strongly recommended that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations, and can thus result in unexpected InvalidClassExceptions during deserialization. Therefore, to guarantee a consistent serialVersionUID value across different java compiler implementations, a serializable class must declare an explicit serialVersionUID value. It is also strongly advised that explicit serialVersionUID declarations use the private modifier where possible, since such declarations apply only to the immediately declaring class--serialVersionUID fields are not useful as inherited members. Array classes cannot declare an explicit serialVersionUID, so they always have the default computed value, but the requirement for matching serialVersionUID values is waived for array classes.

Reply:
Yes, it does not look that `SetBuildDescriptionCommand ` actually needs to be serializable for CLI needs. It looks to be a leftover frrom https://github.com/jenkinsci/jenkins/commit/8b92f50f216a4362c0834635c46f9aa4d8065713 .  I would suggest supressing the warning and adding a TODO comment to remove `Serializable`. Formally it is a breaking change, so keeping it as is would be the safest option. Otherwise we should file it as a separate PR